tests/tests/test_mender_connect.py

```diff
@@ -191,6 +191,57 @@ def test_bogus_shell_message(self, docker_env):
             assert prot.protoType == proto_shell.PROTO_TYPE_SHELL
             assert prot.typ == "bogusmessage"
 
+    @pytest.mark.xfail(reason="CE-277")
+    def test_in_poor_network_environment(self, docker_env):
+        self.assert_env(docker_env)
+
+        breakpoint()
+
+        receive_timeout_s = 16
+
+        def is_shell_is_working(shell):
+            # Test if a simple command works.
+            shell.sendInput("ls /\n".encode())
+            output = shell.recvOutput(receive_timeout_s)
+            assert shell.protomsg.props["status"] == protomsg.PROP_STATUS_NORMAL
+            output = output.decode()
+            assert "usr" in output
+            assert "etc" in output
+
+            breakpoint()
+
+        with docker_env.devconnect.get_websocket() as ws:
+            shell = proto_shell.ProtoShell(ws)
+            body = shell.startShell()
+            assert shell.protomsg.props["status"] == protomsg.PROP_STATUS_NORMAL
+            assert body == proto_shell.MSG_BODY_SHELL_STARTED
+
+            # Drain any initial output from the prompt. It should end in either "# "
+            # (root) or "$ " (user).
+            output = shell.recvOutput(receive_timeout_s)
+            assert shell.protomsg.props["status"] == protomsg.PROP_STATUS_NORMAL
+            assert output[-2:].decode() in [
+                "# ",
+                "$ ",
+            ], "Could not detect shell prompt."
+
+            is_shell_is_working(shell)
+
+            docker_env.device.run(
+                "iptables -A OUTPUT -j DROP --destination docker.mender.io -m statistic --mode random --probability 0.10"
```

Comment:
With a statistical packet drop policy, isn't it possible that it accidentally lets a ping packet through and passes that way? Would it be bad to just drop every packet?

Reply:
I thought about it. But the test is for `degraded` networks, and not for completely degraded network policies.

And the test is actually there so that it should pass. 

But you have a point. I'm undecided

Reply:
Went with the pure `Drop` option